rust/libnewsboat/src/configcontainer.rs

```diff
@@ -0,0 +1,1360 @@
+        );
+
+        config_data.insert(
+            "help-title-format".to_string(),
+            ConfigData::new(&gettext("%N %V - Help"), ConfigDataType::Str),
+        );
+
+        config_data.insert(
+            "itemview-title-format".to_string(),
+            ConfigData::new(
+                &gettext("%N %V - Article '%T' (%u unread, %t total)"),
+                ConfigDataType::Str,
+            ),
+        );
+
+        config_data.insert(
+            "searchresult-title-format".to_string(),
+            ConfigData::new(
+                &gettext(
+                    "%N %V - Search results for '%s' (%u unread, %t total)%?F? matching filter '%F'&?",
+                ),
+                ConfigDataType::Str,
+            ),
+        );
+
+        config_data.insert(
+            "selectfilter-title-format".to_string(),
+            ConfigData::new(&gettext("%N %V - Select Filter"), ConfigDataType::Str),
+        );
+
+        config_data.insert(
+            "selecttag-title-format".to_string(),
+            ConfigData::new(&gettext("%N %V - Select Tag"), ConfigDataType::Str),
+        );
+
+        config_data.insert(
+            "urlview-title-format".to_string(),
+            ConfigData::new(&gettext("%N %V - URLs"), ConfigDataType::Str),
+        );
+
+        config_data.insert(
+            "wrap-scroll".to_string(),
+            ConfigData::new("no", ConfigDataType::Bool),
+        );
+
+        ConfigContainer {
+            config_data: Arc::new(Mutex::new(config_data)),
+        }
+    }
+
+    pub fn handle_action(&self, action: &str, params: &[String]) -> Result<(), ConfigHandlerError> {
+        let mut data = self.config_data.lock().unwrap();
+        if let Some(entry) = data.get_mut(action) {
+            if params.is_empty() {
+                return Err(ConfigHandlerError::TooFewParams);
+            }
+
+            if !entry.multi_option && params.len() > 1 {
+                return Err(ConfigHandlerError::TooManyParams);
+            }
+
+            let value = if entry.multi_option {
+                params.join(" ")
+            } else {
+                params[0].clone()
+            };
+
+            match entry.set_value(value) {
+                Ok(_) => Ok(()),
+                Err(msg) => Err(ConfigHandlerError::InvalidParams(msg)),
+            }
+        } else {
+            Err(ConfigHandlerError::InvalidCommand)
+        }
+    }
+
+    pub fn get_configvalue_as_int(&self, key: &str) -> i32 {
+        self.get_configvalue(key).parse().unwrap_or(0)
+    }
+
+    pub fn get_configvalue_as_bool(&self, key: &str) -> bool {
+        let val = self.get_configvalue(key);
+        val == "true" || val == "yes"
+    }
+
+    pub fn get_configvalue_as_filepath(&self, key: &str, path: &mut std::path::PathBuf) -> bool {
+        let val = self.get_configvalue(key);
+        if !val.is_empty() {
+            *path = std::path::PathBuf::from(val);
+            return true;
+        }
+        false
+    }
```

Comment:
The interface here is different from what we had in C++; why? Can't we just return an empty `PathBuf` if key was not found?